httpbp/headers.go

```diff
@@ -0,0 +1,209 @@
+package httpbp
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	// EdgeContextHeader is the key use to get the raw edge context from
+	// the HTTP request headers.
+	EdgeContextHeader = "X-Edge-Request"
+
+	// ParentIDHeader is the key use to get the span parent ID from
+	// the HTTP request headers.
+	ParentIDHeader = "X-Parent"
+
+	// SpanIDHeader is the key use to get the span ID from the HTTP
+	// request headers.
+	SpanIDHeader = "X-Span"
+
+	// SpanFlagsHeader is the key use to get the span flags from the HTTP
+	// request headers.
+	SpanFlagsHeader = "X-Flags"
+
+	// SpanSampledHeader is the key use to get the sampled flag from the
+	// HTTP request headers.
+	SpanSampledHeader = "X-Sampled"
+
+	// TraceIDHeader is the key use to get the trace ID from the HTTP
+	// request headers.
+	TraceIDHeader = "X-Trace"
+)
+
+type headerContextKey int
+
+const (
+	edgeContextContextKey headerContextKey = iota
+	traceIDContextKey
+	parentIDContextKey
+	spanIDContextKey
+	spanFlagsContextKey
+	spanSampledContextKey
+)
```

Comment:
we actually don't need define these consts. When using them to read/write from the context, we can just do:

```
type headerContextKey HeaderContextKey

ctx.Value(headerContextKey(key))
```

This can also avoid the trouble of using the map.